lib/oauth.js

```diff
@@ -135,6 +135,10 @@ exports.v2 = function (settings) {
           return reply(Boom.internal('App was rejected'));
         }
 
+        // Alter the request to be able to force HTTPS.
+        if (settings.forceHttps) {
+            request.connection.info.protocol = 'https';
```

Comment:
Instead, we should pass a new parameter to `location(request)` for `protocol`.  Otherwise we are overwriting request connection information that may not always be safe.